operator/pkg/controllers/master/master.go

```diff
@@ -0,0 +1,152 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package master
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"net"
+
+	"github.com/awslabs/kit/operator/pkg/apis/infrastructure/v1alpha1"
+	"github.com/awslabs/kit/operator/pkg/utils/secrets"
+
+	pkiutil "github.com/awslabs/kit/operator/pkg/pki"
+	certutil "k8s.io/client-go/util/cert"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	masterCACommonName     = "kubernetes"
+	frontProxyCACommonName = "front-proxy-ca"
+)
+
+type Provider struct {
+	kubeClient client.Client
+}
+
+func New(kubeclient client.Client) *Provider {
+	return &Provider{kubeClient: kubeclient}
+}
+
+func (p *Provider) Reconcile(ctx context.Context, controlPlane *v1alpha1.ControlPlane) (err error) {
+
+	// TODO Create a service for master NLB
```

Comment:
I'd avoid referencing details like NLB, since it's the implementation of the service.